.github/workflows/build_and_test.yml

```diff
@@ -522,9 +522,9 @@ jobs:
           - >-
             pyspark-pandas-slow
           - >-
-            pyspark-pandas-connect-part0, pyspark-pandas-connect-part3
+            pyspark-pandas-connect
```

Comment:
Will this change be backported? The alteration of the group name is likely to affect the daily tests of branch-4.0/branch-3.5.

Reply:
I found a long-standing issue in branch-3.5 that the pandas-api-on-connect tests were actually skipped due to module name mismatch.

https://github.com/apache/spark/actions/runs/17974691765/job/51125480596

branch-3.5 expects module names `pyspark_pandas_connect` and `pyspark_pandas_slow_connect` but get `pyspark-pandas-connect-part?`.

I guess this PR happen to be able to reenable the tests in `Build / Python-only (branch-3.5)`.

For branch-4.0, based on the status in 3.5, I think we don't need backport but we need to adjust the module name in this PR.